### PR TITLE
[Fixes #7860]  Replace layer_thumbnail method for maps 

### LINKED ISSUE #7860: Replace layer_thumbnail method for maps
layer_thumbnail method has been replaced only for datasets. We need it also for maps.

See https://github.com/GeoNode/geonode/issues/7787

### PR DESCRIPTION
**Pull request by mattiagiupponi**
- moved `set_thumbnail_from_bbox` to resourcebase instead of `dataset`
- tests moved from `DatasetApi` to `ResourceBaseApi`

References: #7860

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [x] The issue connected to the PR must have Labels and Milestone assigned
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [x] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates
- [ ] Commits adding **new texts** do use gettext and have updated .po / .mo files (without location infos)

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

**Review by giohappy — COMMENTED** [review]
@mattiagiupponi now that you have moved the endpoint to the base root, did you test how it behaves if called on a document resource?

**Comment by mattiagiupponi** [comment]
@giohappy the `set_thumbnail_from_bbox` for now will work only for `Datasets` and `Maps`. This is because internally, the endpoint calls the `create_thumbnail` which calls `_generate_thumbnail_name` that accept only  `Datasets` and `Maps` as resource types.

https://github.com/GeoNode/geonode/blob/aca637c71bb41b8cafba4c1368479086eb5e6377/geonode/thumbs/thumbnails.py#L221-L236

We can refactor the whole thumbnail creation by including in this endpoint also the `Document` and `GeoApp` thumb generation but will require a some effort

**Comment by giohappy** [comment]
@mattiagiupponi probably I wasn't clear. Precisely because that method is only available for `Datasets` and `Maps` I wouldn't have moved it to the `ResourceBase` API, which is meant to cover methods and properties shared by all the resource types. 

And my question was: what happens if a client calls that base API method passing a reference to a document? Does it raise an exception, is the returned response correctly handles, etc.?

**Comment by mattiagiupponi** [comment]
@giohappy I added a test to show what will happen if a document is passed to the API:
https://github.com/GeoNode/geonode/blob/72dda2dc5752d4a954c43b7ee08fcf17a6dc1a32/geonode/base/api/tests.py#L858

It will rase a `400` with a payload like this:
```json
{
    "message": "Thumbnail generation didn't recognize the provided instance: it's neither a Dataset nor a Map."
}
```

File: geonode/base/api/views.py [hunk 1]
```diff
@@ -436,6 +437,9 @@ def set_thumbnail_from_bbox(self, request, resource_id):
         except ResourceBase.DoesNotExist:
             logger.error(f"Resource selected with id {resource_id} does not exists")
             return Response(data={"message": f"Resource selected with id {resource_id} does not exists"}, status=404, exception=True)
+        except ThumbnailError as e:
```
Comment (giohappy) [inline geonode/base/api/views.py:440]: Please @mattiagiupponi let's implement a more specific exception. This is too much generic and we risk to hide other types of errors.

File: geonode/base/api/views.py [hunk 2]
```diff
@@ -428,6 +430,10 @@ def set_perms(self, request, pk=None):
     def set_thumbnail_from_bbox(self, request, resource_id):
         try:
             resource = ResourceBase.objects.get(id=ast.literal_eval(resource_id))
+
+            if isinstance(resource.get_real_instance(), (Document, GeoApp)):
```
Comment (giohappy) [inline geonode/base/api/views.py:434]: @mattiagiupponi sorry for being picky today but wouldn't it be better to do 

`if not isinstance(resource.get_real_instance(), (Document, Map)):`

? It's better to say what we support rather then list what is not supported :)

**Review by giohappy — APPROVED** [review]

**Comment by afabiani** [comment]
@giohappy @mattiagiupponi I'm afraid the backport PRs must be created manually

**Comment by mattiagiupponi** [comment]
> @giohappy @mattiagiupponi I'm afraid the backport PRs must be created manually

3.2.x - https://github.com/GeoNode/geonode/pull/7866
3.3.x - https://github.com/GeoNode/geonode/pull/7865

**Pull request merged**